### PR TITLE
#1-4Task. kupchic.alex

### PR DESCRIPTION
**Pull request by kupchic**
https://travis-ci.org/github/kupchic/js-assignments

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -238,10 +274,11 @@ function encodeToRot13(str) {
  *   isString(new String('test')) => true
  */
 function isString(value) {
-    throw new Error('Not implemented');
+    if (typeof value === 'string' || value instanceof String) {
+        return true;
+    } else return false;
```
Comment (vramaniuk) [inline task/01-strings-tasks.js]: We can just return (typeof value === 'string' || value instanceof String)

Please, see Please, see https://javascript.info/logical-operators

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -73,7 +76,7 @@ function getAverage(value1, value2) {
  *   (-5,0) (10,-10) => 18.027756377319946
  */
 function getDistanceBetweenPoints(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    return Math.sqrt(Math.pow((x2 - x1), 2) + Math.pow((y2 - y1), 2));
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: Please, improve using Math.hypot

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -111,7 +114,7 @@ function getLinearEquationRoot(a, b) {
  *   (0,1) (1,2)     => 0
  */
 function getAngleBetweenVectors(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    return Math.acos(((x1 * x2) + (y1 * y2)) / (Math.sqrt(x1 * x1 + y1 * y1)) * (Math.sqrt(x2 * x2 + y2 * y2)));
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: Please, try to use Math.hypot

File: task/02-numbers-tasks.js [hunk 4]
```diff
@@ -159,8 +162,9 @@ function parseNumberFromString(value) {
  *   3,3,3   => 5.196152422706632
  *   1,2,3   => 3.741657386773941
  */
-function getParallelipidedDiagonal(a,b,c) {
-    throw new Error('Not implemented');
+function getParallelipidedDiagonal(a, b, c) {
+    return Math.sqrt(a * a + b * b + c * c);
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: Please, use Math.hypot

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -337,9 +374,14 @@ function sortDigitNamesByNumericOrder(arr) {
  *   [ 1, 10, 100, 1000 ]  => 1111
  */
 function getItemsSum(arr) {
-   throw new Error('Not implemented');
+   if (arr.length === 0)
+      return 0;
+   else
+      return arr.reduce(function (a, b) {
+         return a + b;
+      });
```
Comment (vramaniuk) [inline task/04-arrays-tasks.js]: We can use only reduce with initial value 0
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -535,8 +602,20 @@ function selectMany(arr, childrenSelector) {
...
+ 
 function getElementByIndexes(arr, indexes) {
-    throw new Error('Not implemented');
+   // let arr2 = arr.flat(Infinity);
+   // return arr2[indexes.reduce((a, b) => {
+   //    return a+b;
+   // })];
+   let result = null;
+   indexes.map((dimension, index) => {
+      if (index != indexes.length - 1) arr = arr[dimension];
```
Comment (vramaniuk) [inline task/04-arrays-tasks.js]: "operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
'0' != 0
and '0' !== 0)"

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Comment by kupchic** [comment]
> Closing the PR as accepted! Thanks!
have you checked my results html\ccs task?

**Comment by vramaniuk** [comment]
As I see, nobody have checked your html yet.
I'm just checking PR's for now

**Pull request closed without merging**